src/transform/src/reduction_pushdown.rs

```diff
@@ -81,8 +85,712 @@ impl ReductionPushdown {
                     });
                 }
 
-                **input = inner.take_dangerous()
+                **input = inner.take_dangerous();
+
+                // visit relation again to see if the reduction can be pushed
+                // down into whatever the map was around
+                self.action(relation)
```

Comment:
This seems to be the opposite of the `relatino.visit_mut_pre()` change up above. It seems like we will now do a quadratic amount of `action` calls here; is that intended?